.github/workflows/build.yml

```diff
@@ -45,10 +48,40 @@ jobs:
           name: ${{ matrix.asset_name }}
           path: dist/${{ matrix.artifact_name }}
 
+  test_run:
+    if: github.event_name == 'push' && github.ref == 'refs/heads/master'
```

Comment:
do this step on push to master

Reply:
I think `github.event_name == 'push'` should be enough because this only run in master